boards/nordic/nrf52840dk/src/main.rs

```diff
@@ -101,6 +102,9 @@ const SPI_MX25R6435F_CHIP_SELECT: Pin = Pin::P0_17;
 const SPI_MX25R6435F_WRITE_PROTECT_PIN: Pin = Pin::P0_22;
 const SPI_MX25R6435F_HOLD_PIN: Pin = Pin::P0_23;
 
+const QDEC_PIN_A: Pin = Pin::P0_02;
+const QDEC_PIN_B: Pin = Pin::P0_29;
```

Comment:
Is this something that's wired up for all 54820dk's, or something specific to a local testing setup? I don't think we want to wire this up by default unless all boards have something quadrature hanging out at these pins.

Reply:
These pins are specific to the local testing setup, however I don't think it is possible to include the QDEC code from main without picking 2 pins. This raises an interesting question, because if the QDEC code is not included in a main somewhere it is likely to rot. Should this pick 2 pins that are not exposed (and therefore unusable)? There is currently no mechanism for selecting pins from userspace AFAIK.

Reply:
I think #1760 has a similar problem; it's code that's only meaningful if you have an external LoRa radio module attached, which no "stock" board will have. As we expand the set of peripherals, we're going to have more and more of these hardware add-ons. It would be nice to have some more structured mechanism to handle this.

Reply:
I suppose we could resolve this temporarily using the mechanism we use for kernel tests -- put the code that would call this in a qdec_test.rs file, which picks pins and sets up the qdec peripheral and creates a qdec capsule. Then include this file with an `#[allow(unused_code)]` annotation so that code is always built, but not actually run unless it is included from main. The downside is that there is still a lot of friction to actually using it in main, because it will have to be added to the syscall mapping and the Platform struct, and the interrupt mapping will have to be made.